frontend/src/components/NewPassword/NewPassword.js

```diff
@@ -0,0 +1,105 @@
+/* global sessionStorage */
+/* eslint no-undef: "error" */
+
+import React, { Component } from "react";
+import { loginUser } from "../../lib/apiClient";
+import "./NewPassword.css";
+
+class SignIn extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isPasswordVisible: false,
+      email: "",
+      password: ""
+    };
+  }
+
+  toggleIcon = () => {
+    this.setState(prevState => ({
+      isPasswordVisible: !prevState.isPasswordVisible
+    }));
+  };
+
+  handleChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = (event, changeAuthorizedState) => {
+    event.preventDefault();
+    loginUser(this.state)
+      .then(response => {
+        sessionStorage.jwt = response.headers.authorization;
+        changeAuthorizedState(response.data.id);
+      })
+      .catch(error => {
+        if (error.message.includes("401")) {
+          const parentForm = document.getElementsByTagName("form")[1];
+          const errorDiv = document.createElement("p");
+          errorDiv.setAttribute("class", "errorMessage");
+          errorDiv.innerHTML = "Unable to log in. Check your email and password and try again.";
+          parentForm.insertAdjacentElement("beforeend", errorDiv);
+        }
+        console.log(error);
+      });
+  };
```

Comment:
(discussed in person but also adding it here for Henry/Bryan's input) Since this component was copied from SignIn, there are some things that may need to change with these functions. For example, when the submit button is clicked we might not want to automatically sign them in, but instead show some kind of confirmation and then redirect to the sign in page and have them actually sign in with their new password. But, it might be ok to leave some of this in here in case it is still useful for whoever ends up working on the functionality of this component. Not sure what best practice is for a "front end only" ticket-- delete everything that's not relevant, or leave in some stuff that may be useful down the road?

Reply:
I definitely think at least changing some of the text to match the situation (like `"Unable to reset password"` instead of the login error). Also worth noting that there's another PR (#252) that's changing up some of this logic, so it may be easier just to delete all of this stuff (lines 30-44) and come back to it when this gets hooked up to the backend.